integration-webcert/src/main/java/se/inera/intyg/minaintyg/integration/webcert/converter/text/CertificateTextConverter.java

```diff
@@ -0,0 +1,33 @@
+package se.inera.intyg.minaintyg.integration.webcert.converter.text;
+
+import java.util.Collections;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+import se.inera.intyg.minaintyg.integration.api.certificate.model.CertificateLink;
+import se.inera.intyg.minaintyg.integration.api.certificate.model.CertificateText;
+import se.inera.intyg.minaintyg.integration.webcert.client.dto.CertificateTextDTO;
+
+@Component
+@RequiredArgsConstructor
+public class CertificateTextConverter {
+
+  private final CertificateLinkConverter certificateLinkConverter;
+
+  public CertificateText convert(CertificateTextDTO text) {
+    return CertificateText.builder()
+        .text(text.getText())
+        .type(text.getType())
+        .links(
+            convertLinks(text)
+        )
+        .build();
+  }
+
+  private List<CertificateLink> convertLinks(CertificateTextDTO text) {
+    return text.getLinks() == null ? Collections.emptyList()
+        : text.getLinks().stream()
+            .map(certificateLinkConverter::convert)
```

Comment:
Is this converter necessary? Feels like this logic could be moved into CertificateTextConvert

```java
 private List<CertificateLink> convertLinks(CertificateTextDTO text) {
    return text.getLinks() == null ? Collections.emptyList()
        : text.getLinks().stream()
            .map(CertificateTextConverter::convertLink)
            .toList();
  }

  public static CertificateLink convertLink(CertificateLinkDTO link) {
    return CertificateLink.builder()
        .id(link.getId())
        .name(link.getName())
        .url(link.getUrl())
        .build();
  }
```

Reply:
If it's moved in to CertificateText we loose the possibility of converting a CertificateLink by itself in the future. I think having two different converters follows the principle one functionality per service/component. What is negative about having it in to separate converters?

Reply:
Its not negative to implement a separate converter, just felt that it could be handled in CertificateTextConverter :)

If you feel that its appropriate to keep it, feel free to do so